### PR TITLE
[SPARK-25658][SQL][TEST] Refactor HashByteArrayBenchmark to use main method

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Refactor `HashByteArrayBenchmark` to use main method.
1. use `spark-submit`:
```console
bin/spark-submit --class  org.apache.spark.sql.HashByteArrayBenchmark --jars ./core/target/spark-core_2.11-3.0.0-SNAPSHOT-tests.jar ./sql/catalyst/target/spark-catalyst_2.11-3.0.0-SNAPSHOT-tests.jar
```

2. Generate benchmark result:
```console
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "catalyst/test:runMain org.apache.spark.sql.HashByteArrayBenchmark"
```

## How was this patch tested?

manual tests

### CODE REVIEW

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

File: sql/catalyst/src/test/scala/org/apache/spark/sql/HashByteArrayBenchmark.scala [hunk 1]
```diff
@@ -19,15 +19,22 @@ package org.apache.spark.sql
...
 import org.apache.spark.unsafe.Platform
 import org.apache.spark.unsafe.hash.Murmur3_x86_32
 
 /**
  * Synthetic benchmark for MurMurHash 3 and xxHash64.
+ * To run this benchmark:
+ * {{{
+ *   1. without sbt: bin/spark-submit --class <this class> <spark sql test jar>
+ *   2. build/sbt "sql/test:runMain <this class>"
+ *   3. generate result: SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain <this class>"
```
[the remark this answers is not in the document]
Comment (wangyum) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/HashByteArrayBenchmark.scala]: You are right. Thanks

[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/HashByteArrayBenchmark.scala]: Also, please check line 31, too.

**Comment by dongjoon-hyun** [comment]
Could you review and merge https://github.com/wangyum/spark/pull/17 ?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/HashByteArrayBenchmark.scala [hunk 2]
```diff
@@ -19,15 +19,23 @@ package org.apache.spark.sql
...
+import org.apache.spark.benchmark.{Benchmark, BenchmarkBase}
 import org.apache.spark.sql.catalyst.expressions.{HiveHasher, XXH64}
 import org.apache.spark.unsafe.Platform
 import org.apache.spark.unsafe.hash.Murmur3_x86_32
 
 /**
  * Synthetic benchmark for MurMurHash 3 and xxHash64.
+ * To run this benchmark:
+ * {{{
+ *   1. without sbt: bin/spark-submit --class <this class> <spark catalyst test jar>
```
[the remark this answers is not in the document]
Comment (wangyum) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/HashByteArrayBenchmark.scala]: Yes, you are right:
```
LM-SHC-16502798:spark yumwang$ bin/spark-submit --class  org.apache.spark.sql.HashByteArrayBenchmark ./sql/catalyst/target/spark-catalyst_2.11-3.0.0-SNAPSHOT-tests.jar18/10/07 07:35:09 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/spark/benchmark/BenchmarkBase
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
......
```
The correct usage should be:
```
bin/spark-submit --class  org.apache.spark.sql.HashByteArrayBenchmark --jars ./core/target/spark-core_2.11-3.0.0-SNAPSHOT-tests.jar ./sql/catalyst/target/spark-catalyst_2.11-3.0.0-SNAPSHOT-tests.jar
```

**Comment by wangyum** [comment]
retest this please

**Review by dongjoon-hyun — APPROVED** [review]
Thank you, @wangyum .

+1, LGTM. Merged to master.

**Pull request closed without merging**